main.tf

```
@@ -1,9 +1,36 @@
 resource "azurerm_databricks_workspace" "this" {
-  name                        = var.workspace_name
-  resource_group_name         = var.resource_group_name
-  location                    = var.location
-  sku                         = var.sku
-  managed_resource_group_name = var.managed_resource_group_name
+  name                                                = var.workspace_name
+  resource_group_name                                 = var.resource_group_name
+  location                                            = var.location
+  access_connector_id                                 = var.access_connector_id
+  customer_managed_key_enabled                        = var.customer_managed_key_enabled
+  default_storage_firewall_enabled                    = var.default_storage_firewall_enabled
+  infrastructure_encryption_enabled                   = var.infrastructure_encryption_enabled
+  load_balancer_backend_address_pool_id               = var.load_balancer_backend_address_pool_id
+  sku                                                 = var.sku
+  managed_disk_cmk_key_vault_id                       = var.managed_disk_cmk_key_vault_id
+  managed_disk_cmk_key_vault_key_id                   = var.managed_disk_cmk_key_vault_key_id
+  managed_disk_cmk_rotation_to_latest_version_enabled = var.managed_disk_cmk_rotation_to_latest_version_enabled
+  managed_services_cmk_key_vault_id                   = var.managed_services_cmk_key_vault_id
+  managed_services_cmk_key_vault_key_id               = var.managed_services_cmk_key_vault_key_id
+  managed_resource_group_name                         = var.managed_resource_group_name
+  network_security_group_rules_required               = var.network_security_group_rules_required
+  public_network_access_enabled                       = var.public_network_access_enabled
+
+  custom_parameters {
+    machine_learning_workspace_id                        = var.machine_learning_workspace_id
+    nat_gateway_name                                     = var.nat_gateway_name
+    public_ip_name                                       = var.public_ip_name
+    no_public_ip                                         = var.no_public_ip
+    public_subnet_name                                   = var.public_subnet_id != null ? local.public_subnet_parsed.subnet_name : null
+    public_subnet_network_security_group_association_id  = var.public_subnet_network_security_group_association_id
+    private_subnet_name                                  = var.private_subnet_id != null ? local.private_subnet_parsed.subnet_name : null
+    private_subnet_network_security_group_association_id = var.private_subnet_network_security_group_association_id
```

According to the provider docs, the value of `*_subnet_network_security_group_association` is required if `virtual_network_id` is set, and must be the ID of the subnet specified in the corresponding `*_subnet_name` argument, shouldn't we be able to do something like this?

```suggestion
    public_subnet_name                                   = var.public_subnet_name
    public_subnet_network_security_group_association_id  = var.virtual_network_id != null ? "${var.virtual_network_id}/subnets/${var.public_subnet_name}" : null
    private_subnet_name                                  = var.private_subnet_id
    private_subnet_network_security_group_association_id = var.virtual_network_id != null ? "${var.virtual_network_id}/subnets/${var.private_subnet_name}"
```

---

Had a quick discussion face-to-face and decided to keep the explicit configuration for now. Can be simplified in the future if wanted.